### PR TITLE
Add weight / vsize

### PR DESCRIPTION
**Pull request by runn1ng**
I will add some simple tests if this concept is OK with you

### CODE REVIEW

**Comment by runn1ng** [comment]
Hm, `vsize` name is from bitcoin, but it clashes with the naming conventions in the library. `virtualSize` might be better

**Comment by dcousens** [comment]
`virtualSize` :+1: 

LGTM
Tests!

**Comment by runn1ng** [comment]
Done, the tests seem to be going well on Travis.

The fixtures have vsize that I took by decoding from bitcoind, I took weight as vsize*4 and fiddled with it until it passed (since bitcoind doesn't give me weight).

File: test/transaction.js [hunk 1]
```diff
@@ -129,6 +129,24 @@ describe('Transaction', function () {
     })
   })
 
+  describe('weight/virtualSize', function () {
+    it('computes virtual size correctly', function () {
```
Comment (dcousens) [inline test/transaction.js]: quick small nit, `correctly` is implied

**Comment by dcousens** [comment]
Merge after nit

File: test/transaction.js (hunk 1 above)
Comment (runn1ng) [inline test/transaction.js]: done

Comment (runn1ng) [inline test/transaction.js]: ...ok now done :D

**Comment by dcousens** [comment]
Thanks @runn1ng , will release tomorrow (when I wake up)

**Comment by runn1ng** [comment]
I am now thinking, maybe segwit2x and the like define their own transaction weight differently? :)) should that be in the code?

**Comment by runn1ng** [comment]
...no, from btc1/bitcoin code it seems the logic is the same.

I vaguely remember one of these alternative forks defining weight a different way, but maybe I confused all the agreements and proposals together. :D

**Comment by dcousens** [comment]
No unit test failures,  only integration issues (testnet wallet empty?).

**Comment by dcousens** [comment]
Published as `3.1.0`

**Comment by runn1ng** [comment]
yep, seems like it's empty. Plus coverage seems to do some strange stuff on travis

**Comment by junderw** [comment]
https://github.com/bitcoinjs/bitcoinjs-lib/blob/9a8552acda9c1881c43b3b6a9b42fb6b46ebd424/src/transaction.js#L194-L206

```javascript
const bitcoin = require('bitcoinjs-lib')
let tx = bitcoin.Transaction.fromHex("020000000001040f21c6ece2b803811164c37230e2a136d2b021502fd52358f928972e22b7e0c400000000171600143180886204d71c67968bcf0b9ab9aaf2e6d3b3affeffffff36292fab7eee2a4e5790db40d976b4ab5d52c0dab3a3f2b67fdb67c8b74d839000000000171600148232ded34c53cdcf300013d60662adf01f23b9b1feffffffb07862348d0da1f0246d6cc1cff60f790d7ca68e11a43a076c6ff4d61fb9b8861900000017160014bdc3f11aa213678cd20c2018571f7cbcadc808ddfeffffffecbbde3c74a7002a03c40d07104c24d5b2f4e7a7d68a3f7db7c62de46d781d870000000017160014c7ba090770e3aacaa3db7d7038b6dc567cf48b06feffffff0204b18a01000000"+
"001976a914b1c4b7e2109c664c75604220bf659e7e3bc0eb9788ac0a400f000000000017a91435969b509793ba10997a405bbf1933f7a952934887024730440220317534131a88e12f8e2382eaf994f2f7184c36b79af1b6a12c718aa8e0067098022074a916f7b4126c9fa9649306ba1cc4a813586c1d5dc49c669276855e63e963dd012103f415837ddb30cbb4b26078ef558d8cd1283d544b55589126178898a28b36d34c024730440220492447c8d7eba7c53d3b15e66a8989f7e18505c8281b015780439a8b68b9e96602200cd84625feed4404c36d24ba159b668c21b4ceb5c687399a6461"+
"82f2cdf70b3f012103cc990f33df2cbfdf2ed608c0396c2efe58a9796faedb06f7bb2de69e3aac0e0402473044022035a70be8afde5cdb53e762aebb762374dfd5aba82c8e01f4582cbc84b519e1b002204d4203e88de45e3e9a30ef659191108999beedba95ddeef0ad2e6f60d48044e00121030faeaa2837bc33284268ab2bf8e4bd0d7d277f416c22a80fe4b4498bac92031a02483045022100cd6ef572f0d378bfcbf09742424edbd4376310403c8ee7f10c76bdb92c291a8202202625f6bed"+
"011476830d25e7b00f511d80efde5a4ad3febbdb2c9e4660b9c40c1012102d825668ddda77013f1a9f235f1540105d2f98c362831486598b97d08522d5293256e0800")
console.log(tx.weight())
// 1759 weight units. The block size is now capped to 4,000,000 weight units.
console.log(tx.virtualSize())
// 440 virtual Bytes. (weight / 4) This is the "byte size" that is used for calculating fees.
console.log(tx.byteLength())
// 763 bytes. This is the actual number of bytes written on your harddisk to save this transaction.
```

Also keep in mind that the result of a `txb.build()` is a Transaction object, so you can use the above functions.

**Pull request merged**